src/core/compact_object.cc

```diff
@@ -1255,4 +1232,28 @@ MemoryResource* CompactObj::memory_resource() {
   return tl.local_mr;
 }
 
+std::string_view CompactObjTypeConverter::ObjTypeToString(CompactObjType type) {
+  auto it = kObjTypeToString.find(type);
+  if (it != kObjTypeToString.end()) {
+    return it->second;
+  }
+  LOG(ERROR) << "Unsupported type " << type;
```

Comment:
this was a DCHECK(false) = LOG(DFATAL) before

Reply:
actually it was `ObjTypeName` function, that I refactored

Reply:
I think LOG(ERROR) is better for this case, isn't it?

Reply:
Ah, my bad, sorry

Reply:
> I think LOG(ERROR) is better for this case, isn't it?

It won't crash a debug build, so you won't uncover bugs

Reply:
we usually need to log something when a process reaches problematic state.
ObjTypeName logged DFATAL meaning that program is not supposed to reach this state and if it reaches, it's a bug. That implies that precondition is that the argument is valid.

We do not introduce internal APIs that check validity of argument in all functions - it would be a nightmare. 

So as long as it is expected that `type` is valid when calling this function, you can use `LOG(DFATAL)` and return `string_view{}` and drop optional wrapper.